### ISSUE TITLE
QuantBook returns incorrect timestamps

### ISSUE THREAD

**Issue by simonsonjack** [QuantConnect/Research#12]
QuantBook returns timestamps for equities data that don't align with proper market dates. Running the code below in the QC research environment will reproduce the problem.

```
%matplotlib inline
# Imports
from clr import AddReference
AddReference("System")
AddReference("QuantConnect.Common")
AddReference("QuantConnect.Jupyter")
AddReference("QuantConnect.Indicators")
from System import *
from QuantConnect import *
from QuantConnect.Data.Custom import *
from QuantConnect.Data.Market import TradeBar, QuoteBar
from QuantConnect.Jupyter import *
from QuantConnect.Indicators import *
from datetime import datetime, timedelta
import matplotlib.pyplot as plt
import pandas as pd

qb = QuantBook()

spy = qb.AddEquity("SPY")
h1 = qb.History(qb.Securities.Keys, 10000, Resolution.Minute)
df_spy = pd.DataFrame(h1.loc["SPY"]["close"])
print(df_spy[:60])
```

The returned data is
```
                      close
time                       
2018-12-18 15:56:00  255.96
2018-12-18 15:57:00  255.54
2018-12-18 15:58:00  255.45
2018-12-18 15:59:00  255.34
2018-12-18 16:00:00  255.14
2018-12-19 09:31:00  255.21
2018-12-19 09:32:00  255.37
2018-12-19 09:33:00  255.63
2018-12-19 09:34:00  255.68
2018-12-19 09:35:00  255.78
```
12/18 is a Friday but 12/19 is a Saturday. This issue **does not** occur when printing 12/28 and 12/31, which are a Friday and Monday, respectively.

```
print(df_spy[2585:2595])

                       close
time                        
2018-12-28 15:56:00  247.610
2018-12-28 15:57:00  247.460
2018-12-28 15:58:00  247.375
2018-12-28 15:59:00  247.690
2018-12-28 16:00:00  247.810
2018-12-31 09:31:00  249.500
2018-12-31 09:32:00  249.510
2018-12-31 09:33:00  249.560
2018-12-31 09:34:00  249.810
2018-12-31 09:35:00  250.030
```

**Comment by mg64ve** [comment]
Hello, I have same issue but with different assets.
Consider the following code:

```
%matplotlib inline
# Imports
from clr import AddReference
AddReference("System")
AddReference("QuantConnect.Common")
AddReference("QuantConnect.Jupyter")
AddReference("QuantConnect.Indicators")
from System import *
from QuantConnect import *
from QuantConnect.Data.Custom import *
from QuantConnect.Data.Market import TradeBar, QuoteBar
from QuantConnect.Jupyter import *
from QuantConnect.Indicators import *
from datetime import datetime, timedelta
import matplotlib.pyplot as plt
import pandas as pd

# Create an instance
qb = QuantBook()

# Select asset data
assets = ["AU200AUD", "UK100GBP", "SPX500USD"]

au = qb.AddCfd("AU200AUD", Resolution.Daily, Market.Oanda)
uk = qb.AddCfd("UK100GBP", Resolution.Daily, Market.Oanda)
ny = qb.AddCfd("SPX500USD", Resolution.Daily, Market.Oanda)

h1 = qb.History(qb.Securities.Keys, 10, Resolution.Daily)

h1.loc[:,'close']
```

I have just ran 5 minutes back and got this:

```
symbol     time               
UK100GBP   2019-08-20 01:00:00    7191.60
           2019-08-21 01:00:00    7117.35
           2019-08-22 01:00:00    7206.65
           2019-08-23 01:00:00    7145.30
           2019-08-24 01:00:00    7037.55
           2019-08-28 01:00:00    7075.15
           2019-08-29 01:00:00    7115.40
           2019-08-30 01:00:00    7191.55
           2019-08-31 01:00:00    7233.15
           2019-09-03 01:00:00    7233.15
           2019-09-04 01:00:00    7268.35
SPX500USD  2019-08-22 19:00:00    2920.45
           2019-08-23 19:00:00    2849.65
           2019-08-25 19:00:00    2822.75
           2019-08-26 19:00:00    2874.85
           2019-08-27 19:00:00    2872.55
           2019-08-28 19:00:00    2883.25
           2019-08-29 19:00:00    2922.15
           2019-08-30 19:00:00    2922.15
           2019-09-01 19:00:00    2907.75
           2019-09-02 19:00:00    2907.75
           2019-09-03 19:00:00    2901.65
AU200AUD   2019-08-23 10:00:00    6487.50
           2019-08-24 10:00:00    6515.50
           2019-08-26 10:00:00    6404.00
           2019-08-27 10:00:00    6435.90
           2019-08-28 10:00:00    6452.90
           2019-08-29 10:00:00    6498.30
           2019-08-30 10:00:00    6546.80
           2019-08-31 10:00:00    6583.90
           2019-09-02 10:00:00    6561.50
           2019-09-03 10:00:00    6561.50
           2019-09-04 10:00:00    6505.50
Name: close, dtype: float64
```

- UK100 has not 2019-09-02 which is monday and there is not reason for that.
- SPX500 has 2019-09-03 which Sunday.
- AU200 has 2019-08-31 which is Saturday

I think is a data retrieval issue.
As far as I know it happens with Oanda only.
Do you have an update?

**Comment by Martin-Molinero** [comment]
- The original reported issue has been discarded as invalid `12/18 is a Friday but 12/19 is a Saturday` -> `12/18` is a Tuesday and `12/19` is a Wednesday - checked with @simonsonjack 

Hi @mg64ve!
- `SPX500 has 2019-09-03 which Sunday` -> `2019-09-03` is a Tuesday

Note the `time` being shown is the end time of the bar in exchange time zone.

- `UK100 has not 2019-09-02 which is monday and there is not reason for that` good catch I've reported the issue at https://www.quantconnect.com/forum/discussion/6668/uk100gbp-daily-the-issue-starts-from-sep-2nd-2019-and-continues-until-sep-2nd-2019/p1, this can be done at https://www.quantconnect.com/data/issues/open

- `AU200 has 2019-08-31 which is Saturday` I believe this is expected since the end time of Fridays data is Saturday.

Closing the issue for now
